docs/functional-areas/p2p_DTR_api.md

```diff
@@ -0,0 +1,129 @@
+
+...
+
+void main() {
+  ...
+  // Start the token ring protocol
+  DTRenableProtocol(topology);
+  ...
+}
+```
+
+In the example above, the topology is defined as having 4 nodes and their ids are 1, 0, 2, 3.
+
+## Feeding incoming packets to the protocol
+In order for the protocol to work, the user must feed the protocol with incoming packets. This is done by calling the function `DTRp2pIncomingHandler` which in the interface for receiving P2P packets. As mentioned above the protocol uses the port 15 of the P2P API, so it automatically checks if the packet is coming from the port 15 and if it is, it handles it. Otherwise it is discarded. Though, the user can implement their own code to handle the rest incoming packets if they want to.
+
+Example usage :
+``` C
+void p2pcallbackHandler(P2PPacket *p){
+	// If the packet is a DTR service packet, then the handler will handle it.
+    DTRp2pIncomingHandler(p);
+
+	// User code for handling other packets ...
+}
+
+...
+
+
+void main(){
+  ...
+  // Register the callback function to handle incoming packets.
+  p2pRegisterCallback(p2pcallbackHandler);
+  ...
+}
+
+```
+
+
+## Data Broadcast
+
+To send data through the protocol, the user must call the function `DTRsendPacket`. 
+  ``` C
+  bool DTRsendPacket(DTRpacket* packet)
+  ```
+This function takes the packet to be sent as a parameter. The packet must be filled wit the data the user wants to send and by defining the size of them. The function returns true if the packet was sent successfully to the DTR (**not to the receiver copter**) and false  otherwise.Keep in mind that the packet is sent asynchronously and the user can continue to use the P2P API while the packet is being sent.It is not necessary to fill the `.source_id`, `message_type`, `packet_size` fields of the packet since they are automatically filled by the API function. After the execution of the function, the new packet is inserted in the queue of the protocol responsible for all the packets to be sent. 
+  
+
+
+  Example Usage:
+  ``` C
+  // Initialize the packet
+  DTRpacket packet;
+
+  // Fill the packet with the data
+  packet.dataSize = 3;
+  packet.data[0] = 0x01;
+  packet.data[1] = 0x02;
+  packet.data[2] = 0x03;
+  packet.allToAllFlag = 1;
+  DTRsendPacket(&packet);
+
+  ```
+
+
+## Receive Data
+
+The user must call the function `DTRgetPacket` to receive a packet from the DTR. 
+``` C
+bool DTRgetPacket(DTRpacket* packet, uint32_t timeout);
+```
+
+The function blocks for the specified time (in milliseconds) until a packet is received.If the user wants to block indefinitely, the timeout parameter must be set to `portMAX_DELAY`. The function returns true if a packet was received and false otherwise. If a packet is received, the packet is filled with the data received. In case it was received, the packet is filled with the data received and the corresponding packet is released from the queue responsible for the reception of the DTR packets.
```

Comment:
There is a bit of a mix of milliseconds and ticks here as `portMAX_DELAY` is in ticks. Maybe change the signature of the function to take ticks?

Reply:
The function was finally changed to accepting ticks